c10/xpu/XPUDeviceProp.h

```diff
@@ -0,0 +1,125 @@
+#pragma once
+
+#include <c10/xpu/XPUMacros.h>
+#include <sycl/sycl.hpp>
+
+namespace c10::xpu {
+
+struct C10_XPU_API DeviceProp {
+  // Returns the device name of this SYCL device.
+  sycl::info::device::name::return_type device_name;
+  // Returns the device type associated with the device.
+  sycl::info::device::device_type::return_type device_type;
+  // Returns the platform info.
+  sycl::info::platform::name::return_type platform_name;
+  // Returns the vendor of this SYCL device.
+  sycl::info::device::vendor::return_type vendor;
+  // Returns a backend-defined driver version as a std::string.
+  sycl::info::device::driver_version::return_type driver_version;
+  // Returns the SYCL version as a std::string in the form:
+  // <major_version>.<minor_version>
+  sycl::info::device::version::return_type version;
+  // Returns true if the SYCL device is available. Otherwise, return false.
+  sycl::info::device::is_available::return_type is_available;
+  // Returns the maximum size in bytes of the arguments that can be passed to a
+  // kernel.
+  sycl::info::device::max_parameter_size::return_type max_param_size;
+  // Returns the number of parallel compute units available to the device.
+  sycl::info::device::max_compute_units::return_type max_compute_units;
+  // Returns the maximum dimensions that specify the global and local work-item
+  // IDs used by the data parallel execution model.
+  sycl::info::device::max_work_item_dimensions::return_type max_work_item_dims;
+  // Returns the maximum number of workitems that are permitted in a work-group
+  // executing a kernel on a single compute unit.
+  sycl::info::device::max_work_group_size::return_type max_work_group_size;
+  // Returns the maximum number of subgroups in a work-group for any kernel
+  // executed on the device.
+  sycl::info::device::max_num_sub_groups::return_type max_num_sub_groups;
+  // Returns a std::vector of size_t containing the set of sub-group sizes
+  // supported by the device.
+  sycl::info::device::sub_group_sizes::return_type sub_group_sizes;
+  // Returns the maximum configured clock frequency of this SYCL device in MHz.
+  sycl::info::device::max_clock_frequency::return_type max_clock_freq;
+  // Returns the default compute device address space size specified as an
+  // unsigned integer value in bits. Must return either 32 or 64.
+  sycl::info::device::address_bits::return_type address_bits;
+  // Returns the maximum size of memory object allocation in bytes.
+  sycl::info::device::max_mem_alloc_size::return_type max_mem_alloc_size;
+  // Returns the minimum value in bits of the largest supported SYCL built-in
+  // data type if this SYCL device is not of device type
```

Comment:
What's the meaning of `this SYCL device is not of device type`?

Reply:
see line 50